### PR TITLE
LC-326 unit tests for pinecone indexer

### PR DESCRIPTION
**Pull request by akul-sethi**

### CODE REVIEW

File: lucille-plugins/lucille-pinecone/src/test/resources/PineconeIndexerTest/two-namespaces.conf [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# this example illustrates how Lucille can handle a simple use case like indexing the contents of a CSV file into Solr
```
Comment (rseitz) [inline lucille-plugins/lucille-pinecone/src/test/resources/PineconeIndexerTest/two-namespaces.conf:1]: please remove this comment from this conf file and the others in this PR

File: lucille-plugins/lucille-pinecone/src/test/resources/PineconeIndexerTest/two-namespaces.conf [hunk 2]
```diff
@@ -0,0 +1,32 @@
+connectors: [
```
Comment (rseitz) [inline lucille-plugins/lucille-pinecone/src/test/resources/PineconeIndexerTest/two-namespaces.conf:1]: It looks like the test passes the config directly to the indexer constructor; the config is never used by the Runner itself -- which is fine. This means that the connectors and pipelines elements are not needed by this test, so they can be removed. As a general principle, configs used for testing should be minimal and not include settings that the tests don't need.

File: lucille-plugins/lucille-pinecone/src/test/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexerTest.java [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+      Mockito.verify(goodConnection, Mockito.times(0)).close();
+      indexerGood.validateConnection();
+      indexerGood.closeConnection();
+      Mockito.verify(goodConnection, Mockito.times(1)).close();
+    }
+  }
+
+
+  @Test
+  public void testUpsertAndUpdateNoNamespacesProvided() throws Exception {
```
Comment (rseitz) [inline lucille-plugins/lucille-pinecone/src/test/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexerTest.java:163]: Design question: should at least one namespace be required? Should the indexer throw an exception if none are provided? If yes, we can leave this test in place for now and create a separate ticket for the new validation functionality.

File: lucille-plugins/lucille-pinecone/src/test/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexerTest.java [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+
+      UpsertRequest namespace1Upsert = upsertRequest.getAllValues().get(0);
+      UpsertRequest namespace2Upsert = upsertRequest.getAllValues().get(1);
+
+      assertEquals("namespace-1", namespace1Upsert.getNamespace());
+      assertEquals("namespace-2", namespace2Upsert.getNamespace());
+
+      assertEquals(2, namespace1Upsert.getVectorsList().size());
+      assertEquals(2, namespace2Upsert.getVectorsList().size());
+
```
Comment (rseitz) [inline lucille-plugins/lucille-pinecone/src/test/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexerTest.java:223]: Can remove this extra space. There are a few other places to do this throughout the file.

File: lucille-plugins/lucille-pinecone/src/test/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexerTest.java [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+      assertEquals("namespace-2", namespace2Request1.getNamespace());
+      assertEquals("namespace-2", namespace2Request2.getNamespace());
+
+      // make sure vectors are correct for each document and namespace
+      assertEquals(doc0ForNamespace1, namespace1Request1.getValuesList());
+      assertEquals(doc1ForNamespace1, namespace1Request2.getValuesList());
+      assertEquals(doc0ForNamespace2, namespace2Request1.getValuesList());
+      assertEquals(doc1ForNamespace2, namespace2Request2.getValuesList());
+
+      // No metadata is provided when doin updates so testing is unecessary
```
Comment (rseitz) [inline lucille-plugins/lucille-pinecone/src/test/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexerTest.java:345]: typo in "doing"

File: lucille-plugins/lucille-pinecone/src/test/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexerTest.java (hunk 4 above)
Comment (akul-sethi) [reply lucille-plugins/lucille-pinecone/src/test/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexerTest.java:219]: done

**Review by rseitz — APPROVED** [review]

File: lucille-plugins/lucille-pinecone/src/test/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexerTest.java (hunk 3 above)
Comment (kyle-kmw) [reply lucille-plugins/lucille-pinecone/src/test/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexerTest.java:161]: Pinecone does not require a namespace so I don't think we should force a requirement. Technically if no namespace is set the docs are sent to the default namespace "".

File: lucille-plugins/lucille-pinecone/src/test/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexerTest.java (hunk 5 above)
Comment (akul-sethi) [reply lucille-plugins/lucille-pinecone/src/test/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexerTest.java]: done

**Review by kyle-kmw — APPROVED** [review]

File: lucille-plugins/lucille-pinecone/src/test/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexerTest.java (hunk 3 above)
Comment (akul-sethi) [reply lucille-plugins/lucille-pinecone/src/test/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexerTest.java:161]: While the pinecone client may send docs to the default namespace if none is provided, I believe the way the indexer is now, if no namespaces are provided in the config, then no documents are indexed. That is why I have this test.

**Pull request merged**